### ISSUE TITLE
npm audit security issue

### ISSUE THREAD

**Issue by cholz** [nodkz/mongodb-memory-server#235]
An npm security occured with the ` https-proxy-agent` dependy.
All automatic builds with npm audit will fail.

```
$ npm audit

                       === npm audit security report ===


                                 Manual Review
             Some vulnerabilities require your attention to resolve

          Visit https://go.npm.me/audit-guide for additional guidance


  High            Machine-In-The-Middle

  Package         https-proxy-agent

  Patched in      >=3.0.0

  Dependency of   mongodb-memory-server

  Path            mongodb-memory-server > mongodb-memory-server-core >
                  https-proxy-agent

  More info       https://nodesecurity.io/advisories/1184

found 1 high severity vulnerability in 885950 scanned packages
  1 vulnerability requires manual review. See the full report for details.
```

**Comment by nodkz** [comment]
Fixed in 5.2.9
Thanks to @kthakar1990

**Comment by nodkz** [comment]
Completely fixed in 5.2.11, also was fixed "maximum call stack size exceeded" for ReplicaSets.
